### PR TITLE
[Search page] : Create advanced filters for organizations

### PR DESCRIPTION
**Pull request by cmoinier**
### Description

This PR introduces two new filters for the producer and publisher. It also changes the name of the current filter "Producer" to "Organization", which is more fitting for the fields it filters on. 

### Screenshots

image

### Quality Assurance Checklist

- [ ] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [ ] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

**This work is sponsored by MEL**.

### CODE REVIEW

**Comment by jahow** [comment]
This will be a breaking change for both configuration and existing links, to keep in mind for the release :)

File: apps/datahub/src/app/home/search/search-filters/search-filters.component.ts [hunk 1]
```diff
@@ -53,12 +53,14 @@ export class SearchFiltersComponent implements OnInit {
...
       getOptionalSearchConfig().ADVANCED_FILTERS || [
-        'publisher',
+        'organization',
         'format',
         'publicationYear',
         'topic',
         'isSpatial',
         'license',
+        'producer',
+        'publisher',
```
Comment (tkohr) [inline apps/datahub/src/app/home/search/search-filters/search-filters.component.ts:63]: I wouldn't add these to the default advanced filters, but just keep the `organization` here, as it is currently the case.

File: libs/feature/search/src/lib/utils/service/fields.service.ts [hunk 2]
```diff
@@ -70,6 +72,18 @@ export class FieldsService {
...
+      this.injector,
+      'asc',
+      'key'
+    ),
+    publisher: new MultilingualSearchField(
+      'distributorOrgForResourceObject',
+      this.injector,
+      'asc',
+      'key'
+    ),
```
Comment (tkohr) [inline libs/feature/search/src/lib/utils/service/fields.service.ts:86]: I remember @jahow proposing `producerOrg` and `publisherOrg` as names to make it more obvious these come from Org objects as well. I personally don't have a strong opinion on this.

**Review by tkohr — COMMENTED** [review]
Thanks, @cmoinier ! I've tested it on the MEL use case, where it works as expected :slightly_smiling_face: 

Don't forget to update the comment in the `default.toml` regarding the filters and at best even the doc https://geonetwork.github.io/geonetwork-ui/main/docs/guide/configure.html#search ;-)

File: apps/datahub/src/app/home/search/search-filters/search-filters.component.ts (hunk 1 above)
Comment (cmoinier) [reply apps/datahub/src/app/home/search/search-filters/search-filters.component.ts:63]: I hesitated in doing so, I will reverse it

File: docs/reference/search-fields.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ GeoNetwork-UI has built-in logic for several search fields, each of them relying
 These fields are used in the following context:
 
 - when building a URL or permalink from several search criteria; these fields will appear as query parameters in the URL, for instance:  
-  `/search?publisher=MyOrg&format=csv&format=excel`
+  `/search?organization=MyOrg&format=csv&format=excel`
 - when specifying advanced filters [in a configuration file](../guide/configure.md#search)
 
 ## Fields
```
Comment (tkohr) [inline docs/reference/search-fields.md:17]: Would be nice to add the two new fields `producerOrg` and `publisherOrg` here as well :-)

**Review by tkohr — APPROVED** [review]

**Pull request merged**